File: test/cpp/test_read_subject_info.cpp
// Your First C++ Program

#include <iomanip>
#include <iostream>

#include "../src/harmonie_reader.h"

using namespace Harmonie;

int main() {
    HarmonieReader reader;
    std::cout << "Test: test_read_sleep_stages" << std::endl;

    std::string filename = "";
    if (filename == "") {
        std::cout << "ERROR: No filename specified" << std::endl;
        return 0;
    }
    
    std::cout << "Opening file... :" << filename << std::endl;
    bool res = reader.openFile(filename);

    if (!res) {
        std::cout << "ERROR Failed to open the file:" << filename << std::endl;
        return 0;
    } else {
        std::cout << "SUCCESS" << std::endl;
    }

    HarmonieReader::SUBJECT_INFO subjectInfo = reader.getSubjectInfo();

    std::cout << "*** Subject info ***" << std::endl;
    std::cout << "id: " << subjectInfo.id << std::endl;
    std::cout << "firstname: " << subjectInfo.firstname << std::endl;
    std::cout << "lastname: " << subjectInfo.lastname << std::endl;
    std::cout << "sex: " << subjectInfo.sex << std::endl;
    std::cout << "birthDate: " << std::setprecision (17) << subjectInfo.birthDate << std::endl;
    std::cout << "creationDate: " << std::setprecision (17) << subjectInfo.creationDate << std::endl;
    std::cout << "age: " << subjectInfo.age << std::endl;
    std::cout << "height: " << subjectInfo.height << std::endl;
    std::cout << "weight: " << subjectInfo.weight << std::endl;
    std::cout << "bmi: " << subjectInfo.bmi << std::endl;
    std::cout << "waistSize: " << subjectInfo.waistSize << std::endl;

    std::cout << "DONE" << std::endl;
    return 0;
}

File: test/python/test_read_subject_info.py

from sys import platform
import sys
sys.path.insert(0, '')

if platform == "linux" or platform == "linux2":
    import build.HarmonieReader as HarmonieReader
elif platform == "darwin":
    import build.HarmonieReader as HarmonieReader
elif platform == "win32":
    import build.Release.HarmonieReader as HarmonieReader

reader = HarmonieReader.HarmonieReader()
print("Test: test_read_sleep_stages")

filename = ""
if filename == "":
    print("ERROR: No filename specified")
    quit()

print(f"Opening file... :{filename}")
res = reader.open_file(filename)

if (not res):
    print(f"ERROR Failed to open the file:{filename}")
    quit()
else:
    print("SUCCESS")

print(f"Reading subject info...")
subject_info = reader.get_subject_info()
print(f"subject info:{subject_info}")
print(f"id: {subject_info.id}")
print(f"firstname: {subject_info.firstname}")
print(f"lastname: {subject_info.lastname}")
print(f"sex: {subject_info.sex}")
print(f"birthDate: {subject_info.birth_date}")
print(f"creationDate: {subject_info.creation_date}")
print(f"age: {subject_info.age}")
print(f"height: {subject_info.height}")
print(f"weight: {subject_info.weight}")
print(f"bmi: {subject_info.bmi}")
print(f"waistSize: {subject_info.waist_size}")

print("DONE")
